### PR TITLE
feat: your services tile

### PR DESCRIPTION
**Pull request by dylanjames-dd**
# DCMAW-10010 |  User sees 'Your services' tile on the home screen

Added "Your Services" tile on the home screen.

This tile opens the `your-services` page on the external browser when it's tapped.

| IMG_0021 | IMG_0022 |
| - | - |
| image 2024-10-07 at 19 14 07 | image 2024-10-07 at 19 15 04 | 

https://github.com/user-attachments/assets/7e313bc4-5a81-4cdd-ac1c-f2232783d23d

https://github.com/user-attachments/assets/5ee86224-3ede-4405-b950-649b4b23139a

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update
- [x] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- [x] Met all accessibility requirements?
    - [x] Checked dynamic type sizes are applied
    - [x] Checked VoiceOver can navigate your new code
    - [x] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [x] Ensure that the code coverage and SonarCloud checks have passed
- [x] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [x] Ran the app to ensure that no regressions have been caused by changes during code review.

### CODE REVIEW

File: Tests/UnitTests/Screens/Home/HomeViewControllerTests.swift [hunk 1]
```diff
@@ -22,16 +22,25 @@ final class HomeTabViewModelTests: XCTestCase {
     }
 }
 
-extension HomeTabViewModelTests {
+extension HomeViewControllerTests {
     func test_title_contents() throws {
```
Comment (obinns-dd) [inline Tests/UnitTests/Screens/Home/HomeViewControllerTests.swift:26]: please can we add tests for the `UITableViewDataSource` methods?

File: OneLogin.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2121,6 +2145,8 @@
 				DEVELOPMENT_TEAM = N8W395F695;
 				GENERATE_INFOPLIST_FILE = YES;
 				INFOPLIST_FILE = "OneLogin-Info.plist";
+				INFOPLIST_KEY_CFBundleDisplayName = "One Login - Debug";
```
Comment (obinns-dd) [inline OneLogin.xcodeproj/project.pbxproj:2148]: I don't think these build settings should be here, are we able to move them into the `xcconfig` files instead?

**Review by obinns-dd — APPROVED** [review]
I'm happy, subject to my comments 🙂

File: Tests/UnitTests/Screens/Home/ServicesTileViewModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        mockAnalyticsService = nil
+        sut = nil
+        
+        didCallButtonAction = false
+        
+        super.tearDown()
+    }
+}
+
+extension ServicesTileViewModelTests {
```
Comment (obinns-dd) [inline Tests/UnitTests/Screens/Home/ServicesTileViewModelTests.swift:32]: Would be great if we can add a quick test for the callback too: 
image

File: Sources/Views/ContentTile/ServicesTileViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    let body: GDSLocalisedString = "app_yourServicesCardBody"
+    let showSeparatorLine: Bool = true
+    let secondaryButtonViewModel: ButtonViewModel
+    let backgroundColour: UIColor? = .secondarySystemGroupedBackground
+    
+    init(analyticsService: AnalyticsService,
+         action: @escaping () -> Void) {
+        let event = LinkEvent(textKey: "app_yourServicesCardLink",
+                              linkDomain: AppEnvironment.yourServicesLink,
+                              external: .false)
```
Comment (obinns-dd) [inline Sources/Views/ContentTile/ServicesTileViewModel.swift:19]: I think this is an external event, it's external to the Mobile Pod at least. Or are we defining external as outside of One Login?

File: Tests/UnitTests/Screens/Home/HomeViewControllerTests.swift (hunk 1 above)
Comment (dylanjames-dd) [reply Tests/UnitTests/Screens/Home/HomeViewControllerTests.swift:26]: Yep, there's a test for each method we override now, let me know if you think more could be done 👍

File: OneLogin.xcodeproj/project.pbxproj (hunk 2 above)
Comment (dylanjames-dd) [reply OneLogin.xcodeproj/project.pbxproj]: Yeah I don't get why Xcode seems to add these back intermittently, removed 🙂

File: Sources/Views/ContentTile/ServicesTileViewModel.swift (hunk 4 above)
Comment (dylanjames-dd) [reply Sources/Views/ContentTile/ServicesTileViewModel.swift:19]: Looks like it's for outside GDS or potentially outside of government? Either way, I think we can be sure this one isn't 🙂 
image 2024-10-07 at 23 36 49

Comment (obinns-dd) [reply Sources/Views/ContentTile/ServicesTileViewModel.swift:19]: works for me - thanks!

**Review by gabby-san-digco — APPROVED** [review]

**Pull request merged**